lua/nvim-treesitter/parsers.lua

```diff
@@ -971,8 +971,8 @@ end
 
 M.reset_cache()
 
-function M.has_parser(lang)
-  local lang = lang or M.get_buf_lang(api.nvim_get_current_buf())
+function M.has_parser(_lang)
+  local lang = _lang or M.get_buf_lang(api.nvim_get_current_buf())
```

Comment:
I would rather remove the "not shadow" rule from .luacheckrc or do `lang = lang or` without the local. I personally think the "not shadow" rule is a anti-pattern since it prevents you from hiding another variable name that's not relevant in this context